compose/rails.md

```diff
@@ -16,50 +16,86 @@ dependencies, you'll need to define exactly what needs to be included in the
 container. This is done using a file called `Dockerfile`. To begin with, the
 Dockerfile consists of:
 
-    FROM ruby:2.3.3
-    RUN apt-get update -qq && apt-get install -y build-essential libpq-dev nodejs
-    RUN mkdir /myapp
-    WORKDIR /myapp
-    ADD Gemfile /myapp/Gemfile
-    ADD Gemfile.lock /myapp/Gemfile.lock
-    RUN bundle install
-    ADD . /myapp
+```none
+FROM ruby:2.3.3
+RUN apt-get update -qq && apt-get install -y build-essential libpq-dev nodejs
+RUN mkdir /myapp
+WORKDIR /myapp
+ADD Gemfile /myapp/Gemfile
+ADD Gemfile.lock /myapp/Gemfile.lock
+RUN bundle install
+ADD . /myapp
+```
 
 That'll put your application code inside an image that will build a container
 with Ruby, Bundler and all your dependencies inside it. For more information on
-how to write Dockerfiles, see the [Docker user
-guide](/engine/tutorials/dockerimages.md#building-an-image-from-a-dockerfile)
+how to write Dockerfiles, see the
+[Docker user guide](/engine/tutorials/dockerimages.md#building-an-image-from-a-dockerfile)
 and the [Dockerfile reference](/engine/reference/builder.md).
 
 Next, create a bootstrap `Gemfile` which just loads Rails. It'll be overwritten
 in a moment by `rails new`.
 
-    source 'https://rubygems.org'
-    gem 'rails', '5.0.0.1'
+```none
+source 'https://rubygems.org'
+gem 'rails', '5.0.0.1'
+```
 
 You'll need an empty `Gemfile.lock` in order to build our `Dockerfile`.
 
-    touch Gemfile.lock
+```bash
+touch Gemfile.lock
+```
+
+Next, create a [secret](/engine/swarm/secrets.md) to store the database
+password. This example uses a semi-random string using the `openssl` command,
+but you could also generate it using a password manager or another mechanism.
 
-Finally, `docker-compose.yml` is where the magic happens. This file describes
+```bash
+openssl rand -base64 20 | docker secret create postgres_pw -
+```
+
+The `docker-compose.yml` file is where the magic happens. This file describes
 the services that comprise your app (a database and a web app), how to get each
 one's Docker image (the database just runs on a pre-made PostgreSQL image, and
 the web app is built from the current directory), and the configuration needed
-to link them together and expose the web app's port.
-
-    version: '3'
-    services:
-      db:
-        image: postgres
-      web:
-        build: .
-        command: bundle exec rails s -p 3000 -b '0.0.0.0'
-        volumes:
-          - .:/myapp
-        ports:
-          - "3000:3000"
-        depends_on:
-          - db
+to link them together and expose the web app's port. In addition, the secret
```

Comment:
"the secret [...] is stored in a secret" is a little confusing when reading.

Maybe this sentence should instead be something describing the compose directive, like: "In addition, the database password secret is specified in a `secrets` section so that it does not need to be stored in clear text" ?